### PR TITLE
Improved Documentation for Image-Filter-Application

### PR DESCRIPTION
**Pull request by chandrakant100**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,43 @@
...
+### `An Application to apply different Filters to different BMPs Images`
+## Table of Contents
+
+1) [An overview](#project)
+2) [Building Image-Filter-Application](#building)
+3) [Running Image-Filter-Application](#running)
+
+## Project
+
+This Project is Associated with Department of Integrated MCA Utkal Uniersity, Bhubaneshwar, Odisha.
```
Comment (Ank1taS) [inline README.md]:
```suggestion
This project is associated with the Department of Integrated MCA, Utkal University, Bhubaneshwar, Odisha.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,43 @@
...
+## Building
+
+To `build Image-Filter-App` Make sure To install make in your System and g++ compiler.<br>
```
Comment (Ank1taS) [inline README.md]:
```suggestion
To `build Image-Filter-App` make sure to install `make` in your system and `g++ compiler`.<br>
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,43 @@
...
+Make installation in windows -> [link](https://www.technewstoday.com/install-and-use-make-in-windows/)<br>
+Make installation in Ubuntu -> [link](https://askubuntu.com/questions/161104/how-do-i-install-make)
```
Comment (Ank1taS) [inline README.md]:
```suggestion
- Make installation in windows -> [link](https://www.technewstoday.com/install-and-use-make-in-windows/)<br>

- Make installation in Ubuntu -> [link](https://askubuntu.com/questions/161104/how-do-i-install-make) 

```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,43 @@
...
+## Project
+
+This Project is Associated with Department of Integrated MCA Utkal Uniersity, Bhubaneshwar, Odisha.
+## Building
+
+To `build Image-Filter-App` Make sure To install make in your System and g++ compiler.<br>
+Make installation in windows -> [link](https://www.technewstoday.com/install-and-use-make-in-windows/)<br>
+Make installation in Ubuntu -> [link](https://askubuntu.com/questions/161104/how-do-i-install-make) 
+
+To Build `Image-Filter-App` go to your project directory and Open our Terminal and just type.
```
Comment (Ank1taS) [inline README.md:18]:
```suggestion
To build `Image-Filter-App`, go to your project directory and open your `Terminal` and type.
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,43 @@
...
+
+1) [An overview](#project)
+2) [Building Image-Filter-Application](#building)
+3) [Running Image-Filter-Application](#running)
+
+## Project
+
+This Project is Associated with Department of Integrated MCA Utkal Uniersity, Bhubaneshwar, Odisha.
+## Building
+
```
Comment (sabyasachi07) [inline README.md:13]: Create a separate section or division from line 14 to 16 and label it as dependencies and prerequisite

File: README.md [hunk 6]
```diff
@@ -1,2 +1,43 @@
...
+Make installation in Ubuntu -> [link](https://askubuntu.com/questions/161104/how-do-i-install-make) 
+
+To Build `Image-Filter-App` go to your project directory and Open our Terminal and just type.
+```shell
+make
+```
+Boom! You have successfully build `Image-Filter-Application`
+## Running
+
++ Currently `Image-Filter-Application` supports 5 Filters, This Filters can be used using following Flags below.<br>
```
Comment (Ank1taS) [inline README.md:25]:
```suggestion
+ Currently `Image-Filter-Application` supports `5 Filters`, these filters can be used using the following flags below.<br>
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,43 @@
...
++ We have several example Images to test these Filters stored in [imgDB](https://github.com/Blaze-Stars/Image-Filter-Application/tree/main/img-db)
```
Comment (Ank1taS) [inline README.md:26]:
```suggestion
+ We have several examples images to test these filters stored in [imgDB](https://github.com/Blaze-Stars/Image-Filter-Application/tree/main/img-db)
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,43 @@
...
+
+| Filters | Flag |
+| ------- | ----- |
+| GrayScale | -g |
+| Blur | -b |
+| Reflect | -r |
+| Edge | -e |
+| Sepia | -s |
+
+To Use any of the Filters :
```
Comment (Ank1taS) [inline README.md:36]:
```suggestion
To use any of the filters :
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,43 @@
...
+```shell
+./filter -g img-db/<pickAnyImage> <YourOutputFileName>.bmp
+```
+### NOTE
+```
+The Above example shows output for grayscale filter, you can change the flags and image to apply any other Filter
```
Comment (Ank1taS) [inline README.md:42]:
```suggestion
The above example shows output for the grayscale filter, you can change the flags and image to apply any other filter 
```

**Review by Ank1taS — CHANGES_REQUESTED** [review]

**Pull request merged**